src/main/java/com/redhat/exhort/providers/GradleProvider.java

```diff
@@ -0,0 +1,205 @@
+ * Concrete implementation of the {@link Provider} used for converting dependency trees
+ * for Gradle projects (gradle.build) into a content Dot Graphs for Stack analysis or Json for
+ * Component analysis.
+ **/
+public final class GradleProvider extends BaseJavaProvider {
+
+  private Logger log = LoggersFactory.getLogger(this.getClass().getName());
+  public GradleProvider() {
+    super(Type.GRADLE);
+  }
+
+  @Override
+  public Content provideStack(final Path manifestPath) throws IOException {
+    Path tempFile = getDependencies(manifestPath);
+    if (debugLoggingIsNeeded()) {
+      String stackAnalysisDependencyTree = Files.readString(tempFile);
+      log.info(String.format("Package Manager Maven Stack Analysis Dependency Tree Output: %s %s", System.lineSeparator(), stackAnalysisDependencyTree));
+    }
+    Map<String, String> propertiesMap = extractProperties(manifestPath);
+
+    var sbom = buildSbomFromTextFormat(tempFile, propertiesMap, "runtimeClasspath");
+    return new Content(sbom.getAsJsonString().getBytes(), Api.CYCLONEDX_MEDIA_TYPE);
+  }
+
+  private static Path getDependencies(Path manifestPath) throws IOException {
+    // check for custom gradle executable
+    var gradle = Operations.getCustomPathOrElse("gradle");
+    // create a temp file for storing the dependency tree in
+    var tempFile = Files.createTempFile("exhort_graph_", null);
+    // the command will create the dependency tree in the temp file
+    String gradleCommand = gradle + " dependencies";
+
+    String[] cmdList = gradleCommand.split("\\s+");
+    String gradleOutput = Operations.runProcessGetOutput(Path.of(manifestPath.getParent().toString()), cmdList, null);
+    Files.writeString(tempFile, gradleOutput);
+
+    return tempFile;
+  }
+
+  private Path getProperties(Path manifestPath) throws IOException {
+    Path propsTempFile = Files.createTempFile("propsfile", ".txt");
+    var gradle = Operations.getCustomPathOrElse("gradle");
+    String propCmd = gradle + " properties";
+    String[] propCmdList = propCmd.split("\\s+");
+    String properties = Operations.runProcessGetOutput(Path.of(manifestPath.getParent().toString()), propCmdList, null);
+    // Create a temporary file
+    Files.writeString(propsTempFile, properties);
+
+    return propsTempFile;
+  }
+
+  private Sbom buildSbomFromTextFormat(Path textFormatFile, Map<String, String> propertiesMap, String configName) throws IOException {
+    var sbom = SbomFactory.newInstance(Sbom.BelongingCondition.PURL, "sensitive");
+    String root = getRoot(textFormatFile, propertiesMap);
+
+    var rootPurl = parseDep(root);
+    sbom.addRoot(rootPurl);
+    List<String> lines = extractLines(textFormatFile, configName);
+    String[] array = new String[lines.size()];
+    for (int index = 0; index < array.length; index++) {
+      String line = lines.get(index);
+      line = line.replaceAll("---", "-").replaceAll("    ", "  ");
+      line = line.replaceAll(":(.*):(.*) -> (.*)$", ":$1:$3");
```

Comment:
@olavtar this one suppose to elegantly take the right version in case there was conflict and resolution right? it's a very good technique to do so